### ISSUE TITLE
Compiler Issues

### ISSUE THREAD

**Issue by lnicks** [ucla-oarc-web/WebBlocks#447]
On a fresh clone of my repo I am getting errors from the compiler.

```
Tasks: TOP => default => build
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [rake  -- --config=c:/xampp/htdocs/senatede...]
c:/xampp/htdocs/senatedev/sites/all/themes/senate/Rakefile:34:in `rake'
c:/xampp/htdocs/senatedev/sites/all/themes/senate/Rakefile:33:in `chdir'
c:/xampp/htdocs/senatedev/sites/all/themes/senate/Rakefile:33:in `rake'
c:/xampp/htdocs/senatedev/sites/all/themes/senate/Rakefile:51
Tasks: TOP => default
(See full trace by running task with --trace)
```

CC @chris4ucla

**Comment by lnicks** [comment]
Ruby 1.87

**Comment by lnicks** [comment]
950f4a7 is the latest commit on the package folder within the webblocks theme.

**Comment by ebollens** [comment]
For reference, since the ucla/drupal-senate project is private, the error @lnicks is quoting comes from:

``` ruby
sh "rake #{command} -- --config=#{@configpath} #{@args}"
```

This is using the standard call-forward Rakefile approach from the docs. The error is thus obfuscated one layer out, because it's a Rakefile calling a Rakefile.

An email from @rrocchio suggests the v1.0.09 upgrade could have done this:

> Chris says that no one on her team can compile WebBlocks and Drupal since the last WebBlocks release

I'm not sure this has anything to do with the WebBlocks v1.0.09 release, as it only made one change that even affects the compiler:

> https://github.com/ucla/WebBlocks/compare/v1.0.08...v1.0.09

A full list of the things changed:
- Documentation
- Rake test task names
- Builder options
- Javascript reliability fix
- Extension variable values

Next step was to pull the ucla/drupal-senate repo and try to build WebBlocks.

Turns out the error output above was only partial - the full is as follows:

```
rake  -- --config=/Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/Rakefile-config.rb 
[Dispatcher] Executing task: before_init
[Dispatcher] Executing task: init
[Dispatcher] Executing task: after_init
[Dispatcher] Executing task: before_preprocess
[Dispatcher] Executing task: preprocess
[Dispatcher] Executing task: after_preprocess
[Dispatcher] Executing task: before_link
[Dispatcher] Executing task: link
[Dispatcher] Executing task: after_link
[Dispatcher] Executing task: before_compile
[Dispatcher] Executing task: compile
rake aborted!
Compass compile error: 
directory css/compiled/ 
   create css/compiled/blocks-ie.css 
    error blocks.scss (Line 92: File to import not found or unreadable: inc/directory.
Load paths:
  /Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/src/sass
  /Users/ebollens/.rvm/gems/ruby-1.9.3-p392/gems/compass-0.12.2/frameworks/blueprint/stylesheets
  /Users/ebollens/.rvm/gems/ruby-1.9.3-p392/gems/compass-0.12.2/frameworks/compass/stylesheets
  /Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/package
  /Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/.build-tmp/sass
  Compass::SpriteImporter)

Sass::SyntaxError on line ["92"] of /Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/src/sass/blocks.scss: File to import not found or unreadable: inc/directory.
Load paths:
  /Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/src/sass
  /Users/ebollens/.rvm/gems/ruby-1.9.3-p392/gems/compass-0.12.2/frameworks/blueprint/stylesheets
  /Users/ebollens/.rvm/gems/ruby-1.9.3-p392/gems/compass-0.12.2/frameworks/compass/stylesheets
  /Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/package
  /Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/.build-tmp/sass
  Compass::SpriteImporter
Run with --trace to see the full backtrace
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Logger.rb:58:in `failure'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/WebBlocks.rb:142:in `block (2 levels) in compile_css'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/WebBlocks.rb:134:in `chdir'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/WebBlocks.rb:134:in `block in compile_css'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Logger.rb:50:in `block in task'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Logger.rb:97:in `scope'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Logger.rb:49:in `task'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/WebBlocks.rb:129:in `compile_css'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/WebBlocks.rb:123:in `compile'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/Dispatcher.rb:44:in `block (3 levels) in execute'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/Dispatcher.rb:42:in `each'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/Dispatcher.rb:42:in `block (2 levels) in execute'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Logger.rb:44:in `block in system'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Logger.rb:97:in `scope'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Logger.rb:43:in `system'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/Dispatcher.rb:39:in `block in execute'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/Dispatcher.rb:38:in `each'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Build/Dispatcher.rb:38:in `execute'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/webblocks/package/WebBlocks/lib/Rake/Task/build.rb:13:in `block in <top (required)>'
/Users/ebollens/.rvm/gems/ruby-1.9.3-p392/bin/ruby_noexec_wrapper:14:in `eval'
/Users/ebollens/.rvm/gems/ruby-1.9.3-p392/bin/ruby_noexec_wrapper:14:in `<main>'
Tasks: TOP => default => build
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [rake  -- --config=/Users/ebollens/Sites/dr...]
/Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/Rakefile:34:in `block in rake'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/Rakefile:33:in `chdir'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/Rakefile:33:in `rake'
/Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/Rakefile:51:in `block in <top (required)>'
/Users/ebollens/.rvm/gems/ruby-1.9.3-p392/bin/ruby_noexec_wrapper:14:in `eval'
/Users/ebollens/.rvm/gems/ruby-1.9.3-p392/bin/ruby_noexec_wrapper:14:in `<main>'
Tasks: TOP => default
(See full trace by running task with --trace)
```

This indicates that the problem is _not_ with WebBlocks but rather with the particular `blocks.scss` file within the ucla/drupal-senate site. The error in particular:

```
Sass::SyntaxError on line ["92"] of /Users/ebollens/Sites/drupal-senate/sites/all/themes/senate/src/sass/blocks.scss: File to import not found or unreadable: inc/directory.
```

Pulling the lines referred to here:

``` css
@import "inc/calendar";
@import "inc/directory";
@import "inc/events";
@import "inc/footer";
@import "inc/front";
@import "inc/header";
@import "inc/mega";
@import "inc/news";
@import "inc/node";
@import "inc/search";
@import "inc/slideshow";
@import "inc/wysiwyg";
```

Meanwhile, here's an `ls` of the directory:

```
-rw-r--r--  1 ebollens  staff  1212 Sep 14 08:46 _calendar.scss
-rw-r--r--  1 ebollens  staff  1234 Sep 14 08:46 _events.scss
-rw-r--r--  1 ebollens  staff  2424 Sep 14 08:46 _footer.scss
-rw-r--r--  1 ebollens  staff  1106 Sep 14 08:46 _front.scss
-rw-r--r--  1 ebollens  staff  1217 Sep 14 08:46 _header.scss
-rw-r--r--  1 ebollens  staff  4695 Sep 14 08:46 _mega.scss
-rw-r--r--  1 ebollens  staff  1007 Sep 14 08:46 _news.scss
-rw-r--r--  1 ebollens  staff  3470 Sep 14 08:46 _node.scss
-rw-r--r--  1 ebollens  staff  1214 Sep 14 08:46 _search.scss
-rw-r--r--  1 ebollens  staff  3625 Sep 14 08:46 _slideshow.scss
-rw-r--r--  1 ebollens  staff   537 Sep 14 08:46 _wysiwyg.scss
```

So basically you're trying to include a file that doesn't exist.

Closing this issue as it's not an issue with WebBlocks.

**Comment by lnicks** [comment]
@ebollens , thank you for working on this so quickly.  It's much appreciated, and I'm super sorry the first issue turns out to be my error.  I've replaced the file and merged it into the repo.  

Now I'm receiving a different error.  So I ran `--trace` and this is what I got. I went looking for this possible mixin on my theme, not finding it easily, and I avoided using mixins pretty much completely on my site.  Also, I went ahead and commented out bootstrap like I have been doing, and I got the same result.  

```
rake aborted!
Compass compile error:
directory css/compiled/
   create css/compiled/blocks-ie.css
    error blocks.scss (Line 7 of _background_gradient.scss: Undefined mixin 'gradient-reset-filter'.)

Sass::SyntaxError on line 7 of c: Undefined mixin 'gradient-reset-filter'.
Run with --trace to see the full backtrace
c:/xampp/htdocs/senatedev/sites/all/themes/webblocks/package/WebBlocks/lib/Build/Adapter/../../Logger.rb:58:in `failure'
```

**Comment by lnicks** [comment]
So I'm noticing that the rakefile in my base theme is different from the rakefile in webblocks.  

So now I'm wondering if isn't the rakefile in the actual theme & skeleton.  It occurs to me that if the rakefile.rb has changed, its very possible the subtheme rakefile didn't get those changes.  

Going to try updating the senate theme rakefile with the one in the webblocks repo and see what happens.

Edit: A quick review seems to indicate there aren't any major changes.  Attempting to compile without bootstrap once. more

**Comment by ebollens** [comment]
It works just fine for me to compile now that you fixed the missing file bug.

**Comment by lnicks** [comment]
Had to reclone the repo, but it is working now.  @chris4ucla @rrocchio 

Thanks to @ebollens once again.
